main/src/library/Int64.flix

```diff
@@ -266,6 +266,23 @@ namespace Int64 {
     @Time(1) @Space(1)
     pub def flipBit(x: Int64, position: Int32): Int64 = x ^^^ (1i64 <<< position)
 
+    ///
+    /// Returns the integer binary logarithm of `x`.
+    ///
+    @Time(1) @Space(1)
+    pub def log2(x: Int64): Int32 = highestOneBitPosition(x)
+    ///
+    /// Returns the factorial of `x`.
+    ///
+    @Time(factorial(x)) @Space(1)
+    pub def factorial(x: Int64): Int64 = factorialHelper(x, 1i64)
```

Comment:
I'm sure that it does not work for huge values like `Int64.maxValue()`.
Please tell me if there are better way to treat such values.

Reply:
I looked in java and cpp but neither seem to have factorial on the std lib, perhaps since it easily overflows.